test/e2e/settings/ingress_class.go

```diff
@@ -154,4 +186,148 @@ var _ = framework.IngressNginxDescribe("[Flag] ingress-class", func() {
 				Status(http.StatusNotFound)
 		})
 	})
+
+	ginkgo.It("should use IngressClass when no annotation is present", func() {
```

Comment:
This title does not match with the body, there are test cases here that sets the annotation.